api/index.php

```diff
@@ -0,0 +1,203 @@
+<?php
+$relPath="../pinc/";
+include_once($relPath.'bootstrap.inc');
+include_once($relPath.'User.inc');
+include_once($relPath.'misc.inc');
+include_once('exceptions.inc');
+include_once('v1.inc');
+
+// everything is a JSON response
+header("Content-Type: application/json");
+
+handle_cors_headers();
+
+if($maintenance)
+{
+    throw new ApiException("Site is in maintenance mode");
+}
+
+if(!$api_enabled)
+{
+    throw new ApiException("API is not enabled");
+}
+
+api();
+
+# ---------------------------------------------------------------------------
+
+function api()
+{
+    $username = api_authenticate();
+
+    api_rate_limit($username);
+
+    $query_params = $_GET;
+    $path = explode('/', array_get($query_params, "url", ""));
+    unset($query_params["url"]);
+
+    // if the path ended with a /, pop it off for correct routing
```

Comment:
Is this valid? For example, I'd expect "/projects" (which would list all projects) vs "/projects/" (which should error because you're missing a {projectID}), to behave differently.

Reply:
Ah, yes, that would be a better approach. I'll get this updated.